include/geometric_shapes/bodies.h

```diff
@@ -37,7 +37,7 @@
 #ifndef GEOMETRIC_SHAPES_BODIES_
 #define GEOMETRIC_SHAPES_BODIES_
 
-#if __cplusplus <= 199711L
+#if __cplusplus <= 199711L && ! defined (WIN32)
```

Comment:
Why is this needed if you set __cplusplus?

Reply:
because it was failing in moveit2 wherever the header was included. So _cplusplus flag would have been needed in all the CMake files